app/public/js/init.js

```diff
@@ -1,34 +1,58 @@
-(function() {
-  try {
-    // test for JS features needed to load the main script
-    eval( // eslint-disable-line no-eval
-      '(function() {' +
-        // destructuring binding
-        '({a, b} = {a: 10, b: 20, c: 30, d: 40});' +
-        // const, let
-        'const c1 = 5; let var1 = 5;' +
-        // async functions
-        'async function f1 (){return 9};' +
-        // arrow functions
-        '() => {return 4 + 5};' +
-        // template literal syntax
-        '`${"aaa"}`;' + // eslint-disable-line no-template-curly-in-string
-        // default parameters in function
-        'function f2 (a = 3, b, c=false) {return a}' +
-        // classes
-        'class MyClass {};' +
-        // assign object keys from variable
-        'let objKey = "a"; let obj = {[objKey]: 3};' +
-      '})()'
-    )
+// source: https://gist.github.com/DaBs/89ccc2ffd1d435efdacff05248514f38#gistcomment-1976435
+function playsNice(cb) {
+  const x =
+    '(function() {' +
+      // destructuring binding
+      '({a, b} = {a: 10, b: 20, c: 30, d: 40});' +
+      // const, let
+      'const c1 = 5; let var1 = 5;' +
+      // async functions
+      'async function f1 (){return 9};' +
+      // arrow functions
+      '() => {return 4 + 5};' +
+      // template literal syntax
+      '`${"aaa"}`;' + // eslint-disable-line no-template-curly-in-string
+      // default parameters in function
+      'function f2 (a = 3, b, c=false) {return a}' +
+      // classes
+      'class MyClass {};' +
+      // assign object keys from variable
+      'let objKey = "a"; let obj = {[objKey]: 3};' +
+    '})()'
+
+  if (!window.addEventListener) {
+    cb(false)
+  }
+  const n = document.createElement('script')
+  n.ondone = function(event, s) {
+    // eslint-disable-next-line consistent-this
+    s = this
+    if (s.done) {
+      window.removeEventListener('error', s.ondone, true)
+      if (s.parentNode) {
+        s.parentNode.removeChild(s)
+      }
+      return
+    }
+    this.done = 1
+    cb(!(event && event.error))
```

Comment:
I would change the condition to 
```!(event && (event.error || event.type === 'error' || event.lineno))```
I tried it in IE11 and the returned ErrorEvent has the error property set to undefined :D but there are additional properties provided, like "type" and "lineno" which give away the error, so let's keep it broader.